crates/sui-framework/sources/TxContext.move

```diff
@@ -73,35 +73,31 @@ module Sui::TxContext {
 
     #[test_only]
     /// Create a `TxContext` for testing
-    public fun new(signer: signer, tx_hash: vector<u8>, ids_created: u64): TxContext {
+    public fun new(signer: signer, tx_hash: vector<u8>, epoch: u64, ids_created: u64): TxContext {
         assert!(
             Vector::length(&tx_hash) == TX_HASH_LENGTH,
             Errors::invalid_argument(EBadTxHashLength)
         );
-        TxContext { signer, tx_hash, epoch: 0, ids_created }
+        TxContext { signer, tx_hash, epoch, ids_created }
     }
 
     #[test_only]
     /// Create a `TxContext` for testing, with a potentially non-zero epoch number.
-    public fun new_with_epoch(signer: signer, tx_hash: vector<u8>, epoch: u64, ids_created: u64): TxContext {
-        assert!(
-            Vector::length(&tx_hash) == TX_HASH_LENGTH,
-            Errors::invalid_argument(EBadTxHashLength)
-        );
-        TxContext { signer, tx_hash, epoch, ids_created }
+    public fun new_with_epoch(a: address, epoch: u64, hint: u8): TxContext {
```

Comment:
With this change, the difference between `new` and `new_with_epoch` is no longer the epoch anymore, which seems confusing to me. Not sure what's to achieve here.

Reply:
I needed to be able to create a `TxContext` with certain address and epoch from `TestScenario`. It's simpler and cleaner to provide an address and hint instead of a signer and a vector<u8> so I made this change. The names should definitely be changed. I'll figure out something better. In fact, some other test only functions here are a little messy too. The difference between `new` and `new_from_address` is not just the address. I'll try to refactor that too.